TROUBLESHOOTING.md

```diff
@@ -0,0 +1,155 @@
+
+```shell script
+./profiler.sh start -b 4000000 <pid>
+```
+
+`-b 4000000` is used to increase the frame buffer size as the default is usually too small.
+
+To end profiling and gathering the result you can launch the same command with the `stop` subcommand, this will tells you if the buffer frame was too small. <br/>
+The output is a text file that is not really usable, so let's use our preferred performance representation: The FlameGraph.
+
+```shell script
+./profiler.sh stop -f /tmp/cpu-profile.svg --width 1600 <pid>
+```
+
+It will create an SVG FlameGraph (Async Profiler automatically detect that you ask for a FlameGraph thanks to the svg extension)
+that you can open in your browser (and even zoom inside it by clicking on a frame). <br/>
+The example command will create an SVG of 1600 pixels width.
+
+One of our favorite options is `-s` (or `--simple`) that use simple class names instead of FQDNs, the FlameGraph will be more readable with it, but you will lose the package name. <br/>
+You can also limit the profiling duration by using `-d` (or `--duration`) follow by the duration in seconds.
+
+### Allocation profiling
+
+To start allocation profiling, enter the following command:
+
+```shell script
+./profiler.sh start -b 4000000 -e alloc <pid>
+```
+
+`-e` (or `--event`) allow to give the event to profile, by default it's CPU, here we want allocation profiling.
+
+Then you can stop your profiling the same way as for CPU profiling.
+
+```shell script
+./profiler.sh stop -f /tmp/alloc-profile.svg --width 1600 <pid>
+```
+
+## Profiling application startup with Async Profiler
+
+When you want to profile application startup, you cannot use the command line tool as you need a way to start the profiler with your application.
+
+For this case, the Java agent is the best tool.
+It will start profiling when you start the application, then record the profiling data when the application exits.
+
+Here is some example command lines:
+
+```shell script
+# profile CPU startup
+java -agentpath:/path/to/async-profiler/build/libasyncProfiler.so=start,event=cpu,file=startup-cpu-profile.svg,interval=1000000,width=1600,simple\
+    -jar my-application.jar
+
+# profile allocation startup
+java -agentpath:/path/to/async-profiler/build/libasyncProfiler.so=start,event=alloc,file=/tmp/startup-alloc-profile.svg,interval=1000000,width=1600,simple\
+    -jar my-application.jar
+```
+
+Be careful that short options are not supported inside the agent, you need to use the long version.
+
+By default, Async Profiler sample events each 10ms, for debugging Quarkus startup issue this is too much as we start very fast ;) <br/>
+That's why I configure the profiling interval to 1000000ns (so 1ms).
+
+## Profiling application dev mode with Async Profiler
+
+For profiling Quarkus dev mode, you also need to use the Java agent, you can use it the same way as for production run, but you need to pass the `agentpath` option inside the `jvm.args` system property.
```

Comment:
```suggestion
For profiling Quarkus dev mode, the Java agent is again necessary. It can be used in the same way as for the production application with the exception that `agentpath` option needs to be set via the `jvm.args` system property.
```